### DISCUSSION TITLE
Print time summary for tasks when running in parallel

### DISCUSSION THREAD

**Discussion by t1ernan** [sagiegurari/cargo-make#1137] [Q&A]
Hi!  Just found cargo-make! Very impressive tool and feature rich.

I"m trying to setup a basic CI flow, and enabled parallelisation for my tasks, but I noticed when enabling parallelization the time summary information disappears from the std output.

i.e. when my makefile contains

```toml
[tasks.format]
command = "cargo"
args = ["fmt", "--", "--emit=files"]

[tasks.ci]
workspace = false
run_task = { name = ["format"], parallel = true }
```

and I run it like so
```
cargo make -e CARGO_MAKE_CI=true ci
```

I get the time series info without actually any mention of the `format` task time output. This leads to the build time not equaling the sum of the build steps. Without the parallel flag, it behaves as expected.
```
[cargo-make] INFO - Running Task: ci
[cargo-make] INFO - Running Task: install-rustfmt
[cargo-make] INFO - Running Task: format
[cargo-make] INFO - Execute Command: "cargo" "fmt" "--" "--emit=files"
[cargo-make] INFO - ==================Time Summary==================
[cargo-make] INFO - [Setup Env]:                49.45%     0.14 seconds
[cargo-make] INFO - [Setup Env - Crate Info]:   23.08%     0.06 seconds
[cargo-make] INFO - [Setup Env - Rust]:         20.51%     0.06 seconds
[cargo-make] INFO - [Setup Env - Git]:          5.86%      0.02 seconds
[cargo-make] INFO - [Load Makefiles]:           1.10%      0.00 seconds
[cargo-make] INFO - [Setup Env - Duckscript]:   0.00%      0.00 seconds
[cargo-make] INFO - [Setup Env - CI]:           0.00%      0.00 seconds
[cargo-make] INFO - [Setup Env - Project]:      0.00%      0.00 seconds
[cargo-make] INFO - [Setup Env - Vars]:         0.00%      0.00 seconds
[cargo-make] INFO - ci:                         0.00%      0.00 seconds
[cargo-make] INFO - ================================================
[cargo-make] INFO - Build Done in 0.34 seconds.
```

Since I'm new to cargo-make, I'm not sure if this is just a limitation of running parallel on workspace when workspace is disabled

Thanks!

**Comment by sagiegurari** [comment]
sorry for the really late response. many times i miss the discussion posts.
anyhow, i agree, there is an issue here where parallel tasks run in threads and their data is lost.
i admit i'm not sure how to solve it since when doing parallel, you can put there a list of tasks that run in parallel (thats the point of it) and if i write their output to the above list, it will make little sense and confuse users.
i'm open for suggestions.
